limiter.go

```diff
@@ -144,6 +144,20 @@ func (dl *dialLimiter) AddDialJob(dj *dialJob) {
 	go dl.executeDial(dj)
 }
 
+func (dl *dialLimiter) clearAllPeerDials(p peer.ID) {
+	dl.rllock.Lock()
+	defer dl.rllock.Unlock()
+	delete(dl.waitingOnPeerLimit, p)
+
+	arr := dl.waitingOnFd[:0]
+	for _, j := range dl.waitingOnFd {
+		if j.peer != p {
```

Comment:
These are currently considered "active". We'll need to decrement the activePerPeer counter for each one we remove.